sound/soc/sof/ipc4-topology.c

```diff
@@ -1215,13 +1250,41 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 		struct snd_sof_widget *pipe_widget;
 		struct sof_ipc4_pipeline *pipeline;
 
-		pipe_widget = swidget->spipe->pipe_widget;
-		pipeline = pipe_widget->private;
 		ipc4_copier = (struct sof_ipc4_copier *)swidget->private;
 		gtw_attr = ipc4_copier->gtw_attr;
 		copier_data = &ipc4_copier->data;
 		available_fmt = &ipc4_copier->available_fmt;
 
+		pipe_widget = swidget->spipe->pipe_widget;
+		pipeline = pipe_widget->private;
+
+		if (pipeline->use_chain_dma) {
+			u32 host_dma_id;
+			u32 fifo_size;
+
+			host_dma_id = platform_params->stream_tag - 1;
+			pipeline->msg.primary |= SOF_IPC4_GLB_CHAIN_DMA_HOST_ID(host_dma_id);
```

Comment:
do you need to clear the bits before setting it? Can we have a case when the pipeline is use on different DMA channel?

Reply:
The bits are cleared in sof_ipc4_unprepare_copier_module() [1]. I used to clear them also here where the assignment is done, but in some earlier comment I was told not to.

[1] https://github.com/jsarha/linux-sof/blob/8b1f2414aac5a4d1678ea7de6373a3929604efb0/sound/soc/sof/ipc4-topology.c#L1079